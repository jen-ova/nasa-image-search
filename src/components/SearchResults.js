import React from "react";
import "../styles/SearchResults.css";

function SearchResults({ results }) {
  if (!results.length) {
    return <p>No results</p>;
  } else {
    return (
      <>
        <p>Search Results</p>
        <div className="search-results__cards">
          {results.map((image) => (
            <a href={image} target="_blank" rel="noreferrer" key={image}>
              <img
                key={image}
                className="card-image"
                src={image}
                alt="space-pic"
                data-testid="image"
              />
            </a>
          ))}
        </div>
      </>
    );
  }
}

export default SearchResults;
