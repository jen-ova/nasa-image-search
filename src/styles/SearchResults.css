p {
    padding: 1rem;
}

.search-results__cards {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
}

.card-image {
    margin: 2rem;
    border-radius: 0.5rem;
    float: left;
    width: 15rem;
    height: 10rem; 
    object-fit: cover;
}

.card-image:hover {
    transform: scale(1.2);
}

@media only screen and (max-width: 750px) {
    .images {
      flex: 0 0 45%;
    }
}