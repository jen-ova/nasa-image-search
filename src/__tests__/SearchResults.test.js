import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const images = [
    "https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80",
    "https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80",
    "https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80",
    "https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80",
  ];

  it("renders correctly", () => {
    const { asFragment } = render(<SearchResults results={images} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the expected number of images", () => {
    const { getAllByTestId } = render(<SearchResults results={images} />);
    expect(getAllByTestId("image")).toHaveLength(4);
  });
});
