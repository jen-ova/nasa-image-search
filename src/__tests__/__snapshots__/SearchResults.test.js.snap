// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <p>
    Search Results
  </p>
  <div
    class="search-results__cards"
  >
    <a
      href="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="space-pic"
        class="card-image"
        data-testid="image"
        src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
      />
    </a>
    <a
      href="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="space-pic"
        class="card-image"
        data-testid="image"
        src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
      />
    </a>
    <a
      href="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="space-pic"
        class="card-image"
        data-testid="image"
        src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
      />
    </a>
    <a
      href="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="space-pic"
        class="card-image"
        data-testid="image"
        src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
      />
    </a>
  </div>
</DocumentFragment>
`;
